simulationdataschema/model_method.py

```diff
@@ -34,19 +34,1646 @@
+                AtomsState(chemical_symbol='C', orbitals_state=[OrbitalsState('s'), OrbitalsState('px'), OrbitalsState('py'), OrbitalsState('pz')]),
+                AtomsState(chemical_symbol='H', orbitals_state=[OrbitalsState('s')]),
+            ]
+
+        The relevant orbitals for the TB model are the `'pz'` ones for each `'C'` atom. Then, we define:
+
+            orbitals_ref= [OrbitalState('pz'), OrbitalsState('pz')]
+
+        The relevant atoms information can be accessed from the parent AtomsState sections:
+            atom_state = orbitals_ref[i].m_parent
+            index = orbitals_ref[i].m_parent_index
+            atom_position = orbitals_ref[i].m_parent.m_parent.positions[index]
+        """,
+    )
+
+    def resolve_type(self, logger: BoundLogger) -> Optional[str]:
+        """
+        Resolves the `type` of the `TB` section if it is not already defined, and from the
+        `m_def.name` of the section.
+
+        Args:
+            logger (BoundLogger): The logger to log messages.
+
+        Returns:
+            (Optional[str]): The resolved `type` of the `TB` section.
+        """
+        return (
+            self.m_def.name
+            if self.m_def.name in ["DFTB", "xTB", "Wannier", "SlaterKoster"]
+            else None
+        )
+
+    def resolve_orbital_references(
+        self,
+        model_systems: List[ModelSystem],
+        logger: BoundLogger,
+        model_index: int = -1,
+    ) -> Optional[List[OrbitalsState]]:
+        """
+        Resolves the references to the `OrbitalsState` sections from the child `ModelSystem` section.
+
+        Args:
+            model_systems (List[ModelSystem]): The list of `ModelSystem` sections.
+            logger (BoundLogger): The logger to log messages.
+            model_index (int, optional): The `ModelSystem` section index from which resolve the references. Defaults to -1.
+
+        Returns:
+            Optional[List[OrbitalsState]]: The resolved references to the `OrbitalsState` sections.
+        """
+        model_system = model_systems[model_index]
+
+        # If `ModelSystem` is not representative, the normalization will not run
+        if not model_system.is_representative:
+            logger.warning(
+                f"`ModelSystem`[{model_index}] section was not found to be representative."
+            )
+            return None
+
+        # If `AtomicCell` is not found, the normalization will not run
+        atomic_cell = model_system.atomic_cell[0]
+        if atomic_cell is None:
+            logger.warning("`AtomicCell` section was not found.")
+            return None
+
+        # If there is no child `ModelSystem`, the normalization will not run
+        atoms_state = atomic_cell.atoms_state
+        model_system_child = model_system.model_system
+        if model_system_child is None:
+            logger.warning("No child `ModelSystem` section was found.")
+            return None
```

Comment:
These are the same ubiquitous checks I mentioned before. These could go into separate functions into a higher base class.